test/dlc_tests/sanity/test_security_check.py

```diff
@@ -54,6 +66,297 @@ def run_scan(ecr_client, image):
         raise TimeoutError(f"ECR Scan is still in {scan_status} state. Exiting.")
 
 
+def get_new_image_uri(image):
+    """
+    Returns the new image uri for the image being tested. After running the apt commands, the
+    new image will be uploaded to the ECR based on the new image uri.
+
+    :param image: str
+    :param new_image_uri: str
+    """
+    repository_name = os.getenv("UPGRADE_REPO_NAME")
+    ecr_account = f"{os.getenv('ACCOUNT_ID')}.dkr.ecr.{os.getenv('REGION')}.amazonaws.com"
+    upgraded_image_tag = "-".join(image.replace("/", ":").split(":")[1:]) + "-up"
+    new_image_uri = f"{ecr_account}/{repository_name}:{upgraded_image_tag}"
+    return new_image_uri
+
+
+def run_upgrade_on_image_and_push(image, new_image_uri):
+    """
+    Creates a container for the image being tested. Runs apt update and upgrade on the container
+    and the commits the container as new_image_uri. This new image is then pushed to the ECR. 
+
+    :param image: str
+    :param new_image_uri: str
+    """
+    ctx = Context()
+    docker_run_cmd = f"docker run -id --entrypoint='/bin/bash' {image}"
+    container_id = ctx.run(f"{docker_run_cmd}", hide=True, warn=True).stdout.strip()
+    apt_command = "apt-get update && apt-get upgrade"
+    docker_exec_cmd = f"docker exec -i {container_id}"
+    attempt_count = 0
+    apt_ran_successfully_flag = True
```

Comment:
If the default state of this flag is `True`, then any execution that exits from the while loop in an in incorrect way could leave the flag in its default state. The flag would state that the upgrade ran successfully, while the loop has in fact never succeeded. This could happen especially in cases where errors are ignored and outputs are hidden, as done with `ctx.run(..., hide=True, warn=True)` below.

Please start by setting `apt_ran_successfully_flag` to `False`.

Reply:
Started by setting `apt_ran_successfully_flag` to `False`, thanks!!